subsys/bluetooth/common/dummy.c

```diff
@@ -18,9 +20,15 @@
  * their members are bytes or byte arrays, the size is. They must not be padded
  * by the compiler, otherwise the on-wire packet will not map the packed
  * structure correctly.
+ *
+ * The bt_addr structs are not marked __packed because it's considered ugly. But
```

Comment:
considered ugly by some.

Reply:
Where does this belief stem from? For me, there are two reasons why the structs are not packed, and I may be wrong about the first one (would need to be verified, but it's hard to prove without extensive availability to all possible architectures and compilers):
 - Using __packed might force additional and unnecessary CPU instructions, and possibly usage of unneeded registers when accessing the struct. Or to put it in other words, result in inefficient code.
 - Normally, __packed is used to indicate that it's a protocol struct, i.e. a struct that must have as its binary representation exactly as defined, without added padding. The address structs are kind of hybrid, in that they're used both for logical "host format" structs, as well as for protocol payloads (we could have two different structs, but then again their content would be exactly the same, so unnecessary overhead).

Reply:
This kind of hybrid-ness is what's bothering me.
Since we already are using `bt_addr_le_copy`, we could just change its impl to memcpy the address array and assign the type byte. I haven't checked, but I would be surprised if the code emitted is that much different.

This way we can have the host representation not explicitly packed, and a proper packed addr struct used for (de)serialization.